executor/src/witgen/machines/mod.rs

```diff
@@ -234,6 +235,10 @@ pub struct MachineParts<'a, T: FieldElement> {
     pub identities: Vec<&'a Identity<T>>,
     /// Witness columns relevant to this machine.
     pub witnesses: HashSet<PolyID>,
+    /// Maps an identity ID to the corresponding multiplicity column.
+    /// Only contains identity IDs of identities stored in `connections`.
+    /// Note that multiple identity IDs can map to the same multiplicity column.
+    pub identity_id_to_multiplicity: BTreeMap<u64, PolyID>,
```

Comment:
Does it make sense to move this into `Connection` then?

Reply:
Yes!